LanguageFeatures/Extension-types/static_analysis_member_invocation_A01_t03.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Consider an invocation of the extension type member m on the
+/// receiver expression e according to the extension type declaration V with the
+/// actual type arguments T1, ..., Ts. If the invocation includes an actual
+/// argument part (possibly including some actual type arguments) then call it
+/// args. If the invocation does not include an actual argument part, but it
+/// does include a list of actual type arguments, call it typeArgs. Finally,
+/// assume that V declares the type variables X1, ..., Xs.
+///
+/// Note that if the name of m is a name in the interface of Object (that is,
+/// toString, ==, hashCode, runtimeType, or noSuchMethod), the denoted member is
+/// necessarily a non-extension type member, which determines the static
+/// analysis and dynamic semantics.
+///
+/// @description Checks that members of an `Object` are treated as all other
+/// non-extension type members.
+/// @author sgrekhov22@gmail.com
+/// @issue 53740
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+String log = "";
+
+class A {
+}
+
+class B implements A {
+  @override
+  int get hashCode {
```

Comment:
It would be great if we could verify that a call site where the receiver has type `ET2` _will_ use the signature from `B` for `hashCode`, so it would be nice if this signature could be different from the one in `Object`.

One way to do that could be to return `ET3` which is an extension type with `implements int`, and then check that `ET3 _ = myET2.hashCode;` has no compile-time error.

Similarly for `runtimeType`.

The body probably needs to invoke a constructor of `ET3`, because the whole point is that `int` and `Type` shouldn't be assignable to `ET3`.